File: ios/ReactNativeChat/LottieLoaderViewManager.m
//
//  LottieLoaderViewManager.m
//  ReactNativeChat
//
//  Created by Leland Richardson on 4/25/17.
//  Copyright © 2017 Facebook. All rights reserved.
//

#import "LottieLoaderViewManager.h"

#import <React/RCTBridge.h>
#import <React/RCTView.h>
#import <React/RCTUIManager.h>
#import <Lottie/Lottie.h>
#import "LottieContainerView.h"

@implementation LottieLoaderViewManager {

}

RCT_EXPORT_MODULE(LottieLoader)

- (UIView *)view
{
  return [LottieContainerView new];
}

RCT_EXPORT_VIEW_PROPERTY(sourceName, NSString);

// Exercise:
// expose a `loop` boolean property that can toggle whether or not the animation loops

// Exercise:
// with `loop` false, see if you can figure out how to add an `onFinish` event
// http://facebook.github.io/react-native/docs/native-components-ios.html#events

@end

File: android/app/src/main/java/com/reactnativechat/LottieLoaderViewManager.java
package com.reactnativechat;


import com.airbnb.lottie.LottieAnimationView;
import com.facebook.react.uimanager.SimpleViewManager;
import com.facebook.react.uimanager.ThemedReactContext;

public class LottieLoaderViewManager extends SimpleViewManager<LottieAnimationView> {

  private static final String REACT_CLASS = "LottieLoader";

  @Override
  public String getName() {
    return REACT_CLASS;
  }

  @Override
  public LottieAnimationView createViewInstance(ThemedReactContext context) {
    LottieAnimationView view = new LottieAnimationView(context);
    view.setAnimation("Lottie");
    view.loop(true);
    return view;
  }
}
